influxdb3_write/src/write_buffer/mod.rs

```diff
@@ -2350,6 +2369,125 @@ mod tests {
         assert!(result.is_ok());
     }
 
+    #[tokio::test]
+    async fn write_metrics() {
+        let object_store = Arc::new(InMemory::new());
+        let (buf, metrics) = setup_with_metrics(
+            Time::from_timestamp_nanos(0),
+            object_store,
+            WalConfig::test_config(),
+        )
+        .await;
+        let lines_observer = metrics
+            .get_instrument::<Metric<U64Counter>>(WRITE_LINES_TOTAL_NAME)
+            .unwrap();
+        let bytes_observer = metrics
+            .get_instrument::<Metric<U64Counter>>(WRITE_BYTES_TOTAL_NAME)
+            .unwrap();
+
+        let db_1 = "foo";
+        let db_2 = "bar";
+
+        // do a write and check the metrics:
+        let lp = "\
+            cpu,region=us,host=a usage=10\n\
+            cpu,region=eu,host=b usage=10\n\
+            cpu,region=ca,host=c usage=10\n\
+            ";
+        do_writes(
+            db_1,
+            buf.as_ref(),
+            &[TestWrite {
+                lp,
+                time_seconds: 1,
+            }],
+        )
+        .await;
+        assert_eq!(
+            3,
+            lines_observer
+                .get_observer(&Attributes::from(&[("db", db_1)]))
+                .unwrap()
+                .fetch()
+        );
+        let mut bytes: usize = lp.lines().map(|l| l.len()).sum();
+        assert_eq!(
+            bytes as u64,
+            bytes_observer
+                .get_observer(&Attributes::from(&[("db", db_1)]))
+                .unwrap()
+                .fetch()
+        );
+
+        // do another write to that db and check again for updates:
+        let lp = "\
+            mem,region=us,host=a used=1,swap=4\n\
+            mem,region=eu,host=b used=1,swap=4\n\
+            mem,region=ca,host=c used=1,swap=4\n\
+            ";
+        do_writes(
+            db_1,
+            buf.as_ref(),
+            &[TestWrite {
+                lp,
+                time_seconds: 1,
+            }],
+        )
+        .await;
+        assert_eq!(
+            6,
+            lines_observer
+                .get_observer(&Attributes::from(&[("db", db_1)]))
+                .unwrap()
+                .fetch()
+        );
+        bytes += lp.lines().map(|l| l.len()).sum::<usize>();
+        assert_eq!(
+            bytes as u64,
+            bytes_observer
+                .get_observer(&Attributes::from(&[("db", db_1)]))
+                .unwrap()
+                .fetch()
+        );
+
+        // now do a write that will only be partially accepted to ensure that
+        // the metrics are only calculated for writes that get accepted:
+
+        // the legume will not be accepted, because it contains a new tag,
```

Comment:
As a follow up it would be great to add a `influxdb_write_lines_rejected_total` counter.